### PR TITLE
Namespace overhaul

### PR DESCRIPTION
**Pull request by sroettger**
All challenges go into "default".
All kctf system stuff goes into "kctf-system".

This is much nicer to use. First of all, you see everything kctf related by listing the kctf-system namespace. E.g. not all kctf deployments have kctf in the name. It was not obvious to me that external-dns is part of kctf:
```sh
$kctf-kubectl get deployments --namespace kctf-system 
NAME            READY   UP-TO-DATE   AVAILABLE   AGE
external-dns    0/1     1            0           56m
kctf-operator   1/1     1            1           18m
```

And you can list all challenges, pods etc easily:
```sh
$kctf-kubectl get challenges                         
NAME         HEALTH    STATUS    DEPLOYED   PUBLIC
apache-php   healthy   Running   true       false
bash         healthy   Running   true       false
```

### CODE REVIEW

**Comment by sroettger** [comment]
Deleting is super easy as well. The operator tracks which objects belong to the CRD and garbage collects them.

**Comment by sroettger** [comment]
The makefile is getting more and more simple.
Remaining functionality is:
* local tests etc
* build the docker images and add them to the challenge.yaml
* kubectl apply / delete challenge.yaml

**Comment by sroettger** [comment]
Finally the tests went through.

File: base/challenge-skeleton/challenge.yaml [hunk 1]
```diff
@@ -2,7 +2,6 @@ apiVersion: kctf.dev/v1alpha1
 kind: Challenge
 metadata:
   name: mychal
-  namespace: mychal
```
Comment (sirdarckcat) [inline base/challenge-skeleton/challenge.yaml:5]: samples

**Comment by sirdarckcat** [comment]
does this affect the ability to do RBAC? IIRC namespaces are the level at which we are supposed to do multi-tenancy?

**Comment by sroettger** [comment]
> does this affect the ability to do RBAC? IIRC namespaces are the level at which we are supposed to do multi-tenancy?

You should still be able to set the namespace if you want. It's just optional now.

**Comment by sirdarckcat** [comment]
ok, I see. :) sounds good!

**Review by sirdarckcat — APPROVED** [review]

**Pull request merged**